### PR TITLE
LoginServletTest and Login.jsp Updated to Include Password

### PR DESCRIPTION
**Pull request by Linda-Yang**
I updated login.jsp to check for existing usernames and to check if the password is correct. 
In LoginServletTest, I updated the existing user test to include tests for password. I also removed two tests:
1) bad username because register should take care of checking for bad username inputs
2) new user because the new login.jsp redirects back to the login screen if the username does not exist

### CODE REVIEW

File: src/main/java/codeu/controller/LoginServlet.java [hunk 1]
```diff
@@ -60,26 +60,29 @@ public void doGet(HttpServletRequest request, HttpServletResponse response)
...
+        request.getRequestDispatcher("/WEB-INF/view/login.jsp").forward(request, response);
+      }
     }
-
-    if (!userStore.isUserRegistered(username)) {
-      // TODO: add password
-      User user = new User(UUID.randomUUID(), username, null, Instant.now());
-      userStore.addUser(user);
+    else {
+      request.setAttribute("error", "Invalid password.");
```
Comment (saveen-sahni) [inline src/main/java/codeu/controller/LoginServlet.java]: Should this be invalid username or password?

File: src/main/java/codeu/controller/LoginServlet.java [hunk 2]
```diff
@@ -60,26 +60,29 @@ public void doGet(HttpServletRequest request, HttpServletResponse response)
...
    * This function fires when a user submits the login form. It gets the username from the submitted
-   * form data, and then adds it to the session so we know the user is logged in.
+   * form data, checks that they're valid, and either adds the user to the session so we know the 
+   * user is logged in or shows an error to the user. 
    */
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response)
       throws IOException, ServletException {
     String username = request.getParameter("username");
+    String password = request.getParameter("password");
```
Comment (saveen-sahni) [inline src/main/java/codeu/controller/LoginServlet.java:70]: Probably worth checking this edge case - what happens if a user doesn't input anything for one of these fields? Does request.getParameter() give null or an empty string? The first possibility could lead to null pointers.

File: src/test/java/codeu/controller/LoginServletTest.java [hunk 3]
```diff
@@ -52,45 +52,16 @@ public void testDoGet() throws IOException, ServletException {
...
-
-    loginServlet.doPost(mockRequest, mockResponse);
-
-    Mockito.verify(mockRequest)
-        .setAttribute("error", "Please enter only letters, numbers, and spaces.");
-    Mockito.verify(mockRequestDispatcher).forward(mockRequest, mockResponse);
-  }
-
-  @Test
-  public void testDoPost_NewUser() throws IOException, ServletException {
```
Comment (saveen-sahni) [inline src/test/java/codeu/controller/LoginServletTest.java]: I think one way to think about this is - could the LoginServlet still have to deal with a "new user" as an input? If yes, we should cover the case with a unit test. For example, lets say I tried to log in with a made up username (without registering). How should the LoginServlet respond?

File: src/main/java/codeu/controller/LoginServlet.java (hunk 1 above)
Comment (Linda-Yang) [reply src/main/java/codeu/controller/LoginServlet.java]: ah yes! it should be username my bad

File: src/main/java/codeu/controller/LoginServlet.java (hunk 2 above)
Comment (Linda-Yang) [reply src/main/java/codeu/controller/LoginServlet.java:70]: getParameter returns a null string if the parameter does not exist, I will add a null check for both parameters to deal with this case

File: src/test/java/codeu/controller/LoginServletTest.java (hunk 3 above)
Comment (Linda-Yang) [reply src/test/java/codeu/controller/LoginServletTest.java]: LoginServlet should respond with error! I will add that test into the tester file

**Comment by Linda-Yang** [comment]
I updated LoginServlet and LoginServletTest to reflect the comments made:
1) I added a null check in LoginServlet
2) I added 4 new tests to test for nullPassword, nullUsername, newUser, and wrongPassword

**Review by saveen-sahni — APPROVED** [review]

**Review by lloza1998 — APPROVED** [review]
Good job :)
